### PR TITLE
Maturity Model for Event Log Management (M-21-31) Solution

### PR DESCRIPTION
**Pull request by thbanasi**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

**Comment by v-sabiraj** [comment]
@lior-tamir Please review Playbooks.

**Comment by v-sabiraj** [comment]
@anki-narravula Please review Playbooks.

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## 2) M2131_Data Connector Added or Removed
+This alert is designed to monitor data connector configurations. This alert is triggered when a data connector is added, updated, or deleted.<br>
+## 3) M2131_Asset Stopped Logging (heartbeat)
+This alert is designed to monitor assets within the Maturity Model for Event Log Management (M-21-31) standard. The alert triggers when a monitored asset fails to provide a heartbeat within 24 hours.<br>
+## 4) M2131_Log Analytics Workspace: Active Storage is less than 12 Months
+This alert is designed to monitor log retention within the Maturity Model for Event Log Management (M-21-31) standard. The alert triggers when a log analytics workspace in active storage is configured for less than 1 year.<br>
+## 5) M2131_Event Log Management Posture Changed (Event Logging EL0)
+This alert is desinged to monitor Azure policies aligned with the Maturity Model for Event Log Management (M-21-31) standard. The alert triggers when EL0 policy compliance falls below 70% within a 1 week timeframe.<br>
```
Comment (NikTripathi) [inline Solutions/MaturityModelForEventLogManagementM2131/readme.md:50]: Typo desinged -> designed

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+## 6) M2131_Event Log Management Posture Changed (Basic Event Logging EL1)
+This alert is desinged to monitor Azure policies aligned with the Maturity Model for Event Log Management (M-21-31) standard. The alert triggers when EL1 policy compliance falls below 70% within a 1 week timeframe.<br>
```
Comment (NikTripathi) [inline Solutions/MaturityModelForEventLogManagementM2131/readme.md:52]: Typo desinged -> designed

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+## 7) M2131_Event Log Management Posture Changed (Intermediate Event Logging EL2)
+This alert is desinged to monitor Azure policies aligned with the Maturity Model for Event Log Management (M-21-31) standard. The alert triggers when EL2 policy compliance falls below 70% within a 1 week timeframe.<br>
```
Comment (NikTripathi) [inline Solutions/MaturityModelForEventLogManagementM2131/readme.md:54]: Typo desinged -> designed

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+## 8) M2131_Event Log Management Posture Changed (Advanced Event Logging EL3)
+This alert is desinged to monitor Azure policies aligned with the Maturity Model for Event Log Management (M-21-31) standard. The alert triggers when EL3 policy compliance falls below 70% within a 1 week timeframe.<br>
```
Comment (NikTripathi) [inline Solutions/MaturityModelForEventLogManagementM2131/readme.md:56]: Typo desinged -> designed

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+## 1) Notify Log Management Team
+This Security Orchestration, Automation, & Response (SOAR) capability is designed for configuration with the solution's analytics rules. When analytics rules trigger this automation notifies the log management team of respective details via Teams chat and exchange email. this automation reduces requirements to manually monitor the workbook or analytics rules while increasing response times.<br>
+## 2) Open DevOps Task based on Recommendation
+This Security Orchestration, Automation, & Response (SOAR) capability is designed to create an Azure DevOps Task when an ASC recommendation is triggered. This automation enables a consistent response when resources become unhealthy relative to a predefined recommendation, enabling teams to focus on remediation and improving response times.
+## 3) Open JIRA Ticket based on Recommendation
+This Security Orchestration, Automation, & Response (SOAR) capability is designed to open a Jira issue when an recommendation is unhealthy in Microsoft Defender for Cloud. This automation improves time to response by providing consistant notifications when resources become unhealthy relative to a predefined recommendation.
```
Comment (NikTripathi) [inline Solutions/MaturityModelForEventLogManagementM2131/readme.md:64]: Typo consistant ->consistent

**Review by NikTripathi — CHANGES_REQUESTED** [review]

**Review by NikTripathi — CHANGES_REQUESTED** [review]
Few typos are there in workbook like Microsft -> Microsoft , vulnerabilties -> vulnerabilities , Alined  -> Aliened , Correlatd  -> Correlated , Inferfaces->Interfaces, Actvity ->Activity , Unifed ->Unified , Commerical ->Commercial , DisAllowed ->Disallowed , Symlink ->Simulink , Sentnel ->Sentinel  
Please correct.

**Review by NikTripathi — CHANGES_REQUESTED** [review]
Queries are failing. Please update.
image
image
image
image
image
image
image
image

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md (hunk 2 above)
Comment (thbanasi) [reply Solutions/MaturityModelForEventLogManagementM2131/readme.md:52]: Resolved

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md (hunk 5 above)
Comment (thbanasi) [reply Solutions/MaturityModelForEventLogManagementM2131/readme.md:64]: Resolved, Thanks

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md (hunk 1 above)
Comment (thbanasi) [reply Solutions/MaturityModelForEventLogManagementM2131/readme.md:50]: Resolved, Thanks

File: Solutions/MaturityModelForEventLogManagementM2131/readme.md (hunk 3 above)
Comment (thbanasi) [reply Solutions/MaturityModelForEventLogManagementM2131/readme.md:54]: Resolved, Thanks

**Comment by thbanasi** [comment]
@NikTripathi , thanks for reviewing all updates committed. For the workbook query failures, those panels are expected behavior. Those log sources are required for the compliance framework, but we don't have them in the demo environment so they'll fail out.

**Comment by thbanasi** [comment]
@Anki ***@***.***> & @Lior ***@***.***>,

I wanted to follow up to see if anything is needed for approval on the playbooks in PR#3540

Very Respectfully,

TJ Banasik
CISSP-ISSEP,ISSAP,ISSMP
Senior Program Manager
C&AI, Sovereign Cloud Security Group
***@***.******@***.***>
(202) 658-8148



***@***.***

LinkedIn: In/tjbanasik<http://www.linkedin.com/in/tjbanasik>

**Review by NikTripathi — APPROVED** [review]
Workbook looks good, can merge after analytic rules and hunting queries are approved. Thanks

**Comment by thbanasi** [comment]
@Ajeet Prakash ***@***.***> & @Ashwin ***@***.***>,

Would it be possible to review/approve the analytics+hunting rules in this solution? We are hoping to move to Private Previews next week, mainly because this solution addresses a federal government compliance requirement which has aggressive timelines.

Very Respectfully,

TJ Banasik
CISSP-ISSEP,ISSAP,ISSMP
Senior Program Manager
C&AI, Sovereign Cloud Security Group
***@***.******@***.***>
(202) 658-8148



***@***.***

LinkedIn: In/tjbanasik<http://www.linkedin.com/in/tjbanasik>

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Overall, the queries look good. Below are a few small suggestions:

- There are a few comments in your queries. Comments must be on a separate line, not at the end of a query statement line. Can we please fix that
https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#query
 
- We normally do not have queryFrequency, queryPeriod, triggerOperator, triggerThreshold for hunting queries. It seems we have these in your hunting queries. Can we please change that.
 
- We also need to fix the validation errors that we have currently. It seems those are related to Workbook.
image

**Comment by thbanasi** [comment]
@Ajeet Prakash ***@***.***> & @Nikhil Tripathi (TATA CONSULTANCY SERVICES ***@***.***>,
Thanks so much for the update and for your support. I've addressed respective Analytics/Hunting Query recommendations per your guidance. The npm install error for WorkbooksTemplateValidations wasn't present when I submitted this PR last week and the error logs don't provide any details. What is needed to resolve this check and approve/merge the PR?

Very Respectfully,

TJ Banasik
CISSP-ISSEP,ISSAP,ISSMP
Senior Program Manager
C&AI, Sovereign Cloud Security Group
***@***.******@***.***>
(202) 658-8148



***@***.***

LinkedIn: In/tjbanasik<http://www.linkedin.com/in/tjbanasik>

**Comment by NikTripathi** [comment]
@anki-narravula Please review playbooks.

**Review by anki-narravula — APPROVED** [review]
Playbooks looks good. We can merge

**Comment by thbanasi** [comment]
Team,

Anything needed to approve/merge this PR? Hoping to go to private previews next week

Very Respectfully,

TJ Banasik
CISSP-ISSEP,ISSAP,ISSMP
Senior Program Manager
C&AI, Sovereign Cloud Security Group
***@***.******@***.***>
(202) 658-8148



***@***.***

LinkedIn: In/tjbanasik<http://www.linkedin.com/in/tjbanasik>

**Pull request merged**